### PR TITLE
feat: convert graphql api to pull trace evaluations from db

### LINKED ISSUE #2777: [gql] evals API changes to read from tables
make gql api changes. this most likely will require data loaders as we need to pull evals per span node.

### PR DESCRIPTION
**Pull request by axiomofjoy**
makes graphql api pull trace evaluations from db

resolves #2777

### CODE REVIEW

File: src/phoenix/server/api/types/Project.py [hunk 1]
```diff
@@ -146,10 +146,22 @@ async def latency_ms_quantile(
...
-    def trace(self, trace_id: ID) -> Optional[Trace]:
-        if self.project.has_trace(TraceID(trace_id)):
-            return Trace(trace_id=trace_id, project=self.project)
-        return None
+    async def trace(self, trace_id: ID, info: Info[Context, None]) -> Optional[Trace]:
+        async with info.context.db() as session:
+            if (
+                trace_rowid := await session.scalar(
+                    select(models.Trace.id)
+                    .join(models.Project)
```
Comment (axiomofjoy) [inline src/phoenix/server/api/types/Project.py:154]: When #2839 lands, we should add `project_rowid` as a private field on the `Project` type so we don't have to do this join.

Comment (axiomofjoy) [reply src/phoenix/server/api/types/Project.py:154]: Created issue #2868

File: src/phoenix/server/api/types/Trace.py [hunk 2]
```diff
@@ -44,13 +43,12 @@ async def spans(
             spans = await session.scalars(
                 select(models.Span)
                 .join(models.Trace)
-                .filter(models.Trace.trace_id == self.trace_id)
+                .filter(models.Trace.id == self.trace_rowid)
```
Comment (RogerHYang) [inline src/phoenix/server/api/types/Trace.py:46]: this is my bad (or autocomplete). filter is legacy syntax
```suggestion
                .where(models.Trace.id == self.trace_rowid)
```

**Review by RogerHYang — APPROVED** [review]

File: src/phoenix/server/api/types/Trace.py (hunk 2 above)
Comment (axiomofjoy) [reply src/phoenix/server/api/types/Trace.py:46]: addressed in ea0b8a3ff868118ad08997aebea2efa2c72dfdbb

File: src/phoenix/server/api/types/Project.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/server/api/types/Project.py]: Resolved in this PR

**Pull request merged**